packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -622,6 +624,21 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
       filteredEntries = widget.dropdownMenuEntries;
       buttonItemKeys = List<GlobalKey>.generate(filteredEntries.length, (int index) => GlobalKey());
       _menuHasEnabledItem = filteredEntries.any((DropdownMenuEntry<T> entry) => entry.enabled);
+      if (_selectedEntryIndex != null) {
+        final T oldSelectionValue = oldWidget.dropdownMenuEntries[_selectedEntryIndex!].value;
```

Comment:
this doesn't work because the `_selectedEntryIndex` is relative to the _filtered_ `dropdownMenuEntries`, as set during build.

What this ends up doing is if you have filtering enabled, when you select the nth item from the filtered list, the text field immediately reverts to the nth item in the unfiltered list.

Reply:
Filed https://github.com/flutter/flutter/issues/174609